### PR TITLE
[gitlab] update runners for build jobs

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Migrate Gitlab runners according to new processes

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -48,67 +48,75 @@ generate_code:
 
 build_eds_image_amd64:
   stage: image
-  tags: ["runner:docker", "size:large"]
+  tags: 
+    - "arch:amd64"
```
Comment (celenechang) [inline .gitlab-ci.yml:52]: checking if we ought to use `size:large` tags for the new runners as well

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ variables:
   GOPATH: "$CI_PROJECT_DIR/.cache"
   TARGET_TAG: v$CI_PIPELINE_ID-$CI_COMMIT_SHORT_SHA
   BUILD_DOCKER_REGISTRY: "486234852809.dkr.ecr.us-east-1.amazonaws.com/ci"
-  JOB_DOCKER_IMAGE: "486234852809.dkr.ecr.us-east-1.amazonaws.com/ci-containers-project:v13932656-a1b1db4-v1.0.0"
+  JOB_DOCKER_IMAGE: "486234852809.dkr.ecr.us-east-1.amazonaws.com/docker:24.0.4-gbi-focal"
```
Comment (clamoriniere) [inline .gitlab-ci.yml:9]: maybe you can use https://github.com/DataDog/images/tree/master/ci-containers-project/v2.0.0
I think Charly moved the WPA repo to it

Comment (celenechang) [reply .gitlab-ci.yml:9]: didn't know about that, thanks!

File: .gitlab-ci.yml (hunk 1 above)
Comment (celenechang) [reply .gitlab-ci.yml:52]: from inspecting our runner nodes i don't see any indication of a `size` tag or label

Comment (celenechang) [reply .gitlab-ci.yml:52]: from inspecting our runner nodes i don't see any indication of a `size` tag or label

**Review by levan-m — APPROVED** [review]

**Pull request merged**